### PR TITLE
[PINS] Update Redis schema to match P4Orch

### PR DESCRIPTION
**Pull request by bocon13**
This change reverts some of the modifications made during the review of https://github.com/Azure/sonic-swss-common/pull/536.

### APPL_STATE_DB Separator Rationale

While `APPL_STATE_DB` is a new database from SONiC's perspective, it's intended to mirror `APPL_DB` and thus should used the same separator. If these two databases use different separators, then clients (i.e. P4Orch, P4RT) need to transform these keys when performing the mirroring.

| `APPL_DB` key | `APPL_STATE_DB` key |
| ---------------- | ----------------------- |
| `"VRF_TABLE:vrf-1"` | `"VRF_TABLE\|vrf-1"` |

This difference creates additional complexity and increases the likelihood of bugs. Therefore, we propose using the `APPL_DB` separator in `APPL_STATE_DB`. If/when `APPL_DB` migrates from `:` to `|`, we should migrate  `APPL_STATE_DB` as well.

This is described in greater detail in the Response Path HLD:
https://github.com/pins/SONiC/blob/master/doc/pins/appl_state_db_response_path_hld.md

### APP_P4RT_ Table Names Rationale

The format for P4RT tables in `APPL_DB` uses a two part scheme:
```
<P4RT Prefix> + database separator + <P4RT Table> + P4RT separator + <Table Key>
```

For example,
```
P4RT:FIXED_IPV4_TABLE:{"match/ipv4_dst":"172.16.111.0/24","match/vrf_id":""}

where
P4RT Prefix = P4RT
database separator = :
P4RT Table = FIXED_IPV4_TABLE
P4RT separator = :
Table Key = {"match/ipv4_dst":"172.16.111.0/24","match/vrf_id":""}
```

Everything under P4RT Prefix maps to P4Orch, and the P4RT Table is used to map the entry to the appropriate P4Orch manager.

However, under the merged change we have:
```
In APPL_DB:
P4RT_TABLE:P4RT_IPV4_TABLE:{"match/ipv4_dst":"172.16.111.0/24","match/vrf_id":""}

In APPL_STATE_DB:
P4RT_TABLE|P4RT_IPV4_TABLE:{"match/ipv4_dst":"172.16.111.0/24","match/vrf_id":""}
```

In this case, `_TABLE` in the P4RT Prefix and `P4RT_` in the P4RT Table are both redundant. So, we propose removing those from the defines.

Furthermore we rely on special prefixes to map P4 entities to Redis entries:

- `FIXED_` is used for fixed SAI tables
- `ACL_` is used for programmable tables

Much of the code and tests rely on these prefixes.

This is described in greater detail in the APPL DB Schema HLD:
https://github.com/pins/SONiC/blob/master/doc/pins/P4RT_App_DB_Schema.md

### CODE REVIEW

File: common/schema.h [hunk 1]
```diff
@@ -44,15 +44,16 @@ namespace swss {
...
-#define APP_P4RT_MIRROR_SESSION_TABLE_NAME    "P4RT_MIRROR_SESSION_TABLE"
+#define APP_P4RT_TABLE_NAME                   "P4RT"
+#define APP_P4RT_TABLE_NAME_SEPARATOR         ":"
+#define APP_P4RT_ROUTER_INTERFACE_TABLE_NAME  "FIXED_ROUTER_INTERFACE_TABLE"
+#define APP_P4RT_NEIGHBOR_TABLE_NAME          "FIXED_NEIGHBOR_TABLE"
+#define APP_P4RT_NEXTHOP_TABLE_NAME           "FIXED_NEXTHOP_TABLE"
+#define APP_P4RT_WCMP_GROUP_TABLE_NAME        "FIXED_WCMP_GROUP_TABLE"
+#define APP_P4RT_IPV4_TABLE_NAME              "FIXED_IPV4_TABLE"
+#define APP_P4RT_IPV6_TABLE_NAME              "FIXED_IPV6_TABLE"
+#define APP_P4RT_ACL_TABLE_DEFINITION_NAME    "DEFINITION"
```
[the remark this answers is not in the document]
Comment (bocon13) [reply common/schema.h]: Done

File: common/schema.h [hunk 2]
```diff
@@ -44,15 +44,16 @@ namespace swss {
...
-#define APP_P4RT_ROUTER_INTERFACE_TABLE_NAME  "P4RT_ROUTER_INTERFACE_TABLE"
-#define APP_P4RT_NEIGHBOR_TABLE_NAME          "P4RT_NEIGHBOR_TABLE"
-#define APP_P4RT_NEXTHOP_TABLE_NAME           "P4RT_NEXTHOP_TABLE"
-#define APP_P4RT_WCMP_GROUP_TABLE_NAME        "P4RT_WCMP_GROUP_TABLE"
-#define APP_P4RT_IPV4_TABLE_NAME              "P4RT_IPV4_TABLE"
-#define APP_P4RT_IPV6_TABLE_NAME              "P4RT_IPV6_TABLE"
-#define APP_P4RT_ACL_TABLE_DEFINITION_NAME    "P4RT_ACL_TABLE_DEFINITION"
-#define APP_P4RT_MIRROR_SESSION_TABLE_NAME    "P4RT_MIRROR_SESSION_TABLE"
+#define APP_P4RT_TABLE_NAME                   "P4RT"
+#define APP_P4RT_TABLE_NAME_SEPARATOR         ":"
```
[the remark this answers is not in the document]
Comment (bocon13) [reply common/schema.h]: We already have `kTableKeyDelimiter` defined in p4orch_util.h, so replacing instances of `APP_P4RT_TABLE_NAME_SEPARATOR` with `kTableKeyDelimiter`.

File: common/schema.h [hunk 3]
```diff
@@ -44,15 +44,16 @@ namespace swss {
...
-#define APP_P4RT_TABLE_NAME                   "P4RT_TABLE"
-#define APP_P4RT_ROUTER_INTERFACE_TABLE_NAME  "P4RT_ROUTER_INTERFACE_TABLE"
-#define APP_P4RT_NEIGHBOR_TABLE_NAME          "P4RT_NEIGHBOR_TABLE"
-#define APP_P4RT_NEXTHOP_TABLE_NAME           "P4RT_NEXTHOP_TABLE"
-#define APP_P4RT_WCMP_GROUP_TABLE_NAME        "P4RT_WCMP_GROUP_TABLE"
-#define APP_P4RT_IPV4_TABLE_NAME              "P4RT_IPV4_TABLE"
-#define APP_P4RT_IPV6_TABLE_NAME              "P4RT_IPV6_TABLE"
-#define APP_P4RT_ACL_TABLE_DEFINITION_NAME    "P4RT_ACL_TABLE_DEFINITION"
-#define APP_P4RT_MIRROR_SESSION_TABLE_NAME    "P4RT_MIRROR_SESSION_TABLE"
+#define APP_P4RT_TABLE_NAME                   "P4RT"
```
[the remark this answers is not in the document]
Comment (bocon13) [reply common/schema.h]: Done

**Review by prsunny — APPROVED** [review]

**Comment by bocon13** [comment]
The test failures are unrelated to this change. They are `docker.errors.NotFound` errors in the vs tests.

@prsunny can you please merge if you are okay with the PR?

**Comment by prsunny** [comment]
the merge can be done after all the test passes.

**Comment by bocon13** [comment]
> the merge can be done after all the test passes.

That makes sense. Thanks for restarting the build :)

**Comment by bocon13** [comment]
@bocon13 Rebase and rerun tests

**Pull request merged**